assertj-core/src/main/java/org/assertj/core/api/RecursiveComparisonAssert.java

```diff
@@ -1525,11 +1529,59 @@ public <T> SELF withComparatorForType(Comparator<? super T> comparator, Class<T>
     return myself;
   }
 
+  /**
+   * Allows to register a comparator to compare the fields with the given types.
+   * A typical usage is to compare fields belonging to different types.
+   * <p>
+   * Comparators registered with this method have less precedence than comparators registered with {@link #withComparatorForFields(Comparator, String...) withComparatorForFields(Comparator, String...)}
+   * or BiPredicate registered with {@link #withEqualsForFields(BiPredicate, String...) withEqualsForFields(BiPredicate, String...)}.
+   * <p>
+   * Note that registering a {@link BiComparator} for a given types will override the previously registered BiPredicate/Comparator (if any).
+   * <p>
+   * Example:
+   * <pre><code class='java'> public class TolkienCharacter {
+   *   LocalDate birthday;
+   * }
+   * public class TolkienCharacterDto {
+   *    String birthday;
+   * }
+   *
+   * TolkienCharacter frodo = new TolkienCharacter(LocalDate.of(2968, Month.SEPTEMBER, 22));
+   * TolkienCharacterDto frodoDto = new TolkienCharacterDto("2968-09-22");
+   * TolkienCharacterDto bilboDto = new TolkienCharacterDto("2890-09-22");
+   *
+   * BiComparator&lt;LocalDate, String&gt; sameDate = (d, s) -&gt; LocalDate.parse(s).compareTo(td);
+   *
+   * // assertion succeeds
+   * assertThat(frodo).usingRecursiveComparison()
+   *                  .withComparatorForTypes(sameDate, LocalDate.class, String.class)
+   *                  .isEqualTo(frodoDto);
+   *
+   * // assertion fails
+   * assertThat(frodo).usingRecursiveComparison()
+   *                  .withComparatorForTypes(closeEnough, Double.class)
+   *                  .isEqualTo(bilboDto);</code></pre>
+   *
+   * @param <T> the left element's class type to register a comparator for
+   * @param <U> the right element's class type to register a comparator for
+   * @param comparator the {@link BiComparator Comparator} to use to compare the given fields
+   * @param type the type of the left element to be compared with the given comparator.
+   * @param otherType the type of the right element to be compared with the given comparator.
+   *
+   * @return this {@link RecursiveComparisonAssert} to chain other methods.
+   * @throws NullPointerException if the given comparator is null.
+   */
+  public <T, U> SELF withComparatorForTypes(BiComparator<? super T, ? super U> comparator, Class<T> type, Class<U> otherType) {
```

Comment:
I thought the PR requires also the `withComparatorForTypes` implementation because it's mentioned in the linked issue. If that's not the case, I'll be more than happy to rid of it

Reply:
Go for it !

Reply:
method removed in 122d8af03bf46ff5716bf578d37b0cf6da4bf4aa